js/src/logger.ts

```diff
@@ -867,9 +868,16 @@ class BackgroundLogger {
       this.maxRequestSize = maxRequestSizeEnv;
     }
 
-    const numRetriesEnv = Number(iso.getEnv("BRAINTRUST_NUM_RETRIES"));
-    if (!isNaN(numRetriesEnv)) {
-      this.numRetries = numRetriesEnv;
+    const numTriesEnv = Number(iso.getEnv("BRAINTRUST_NUM_RETRIES"));
+    if (!isNaN(numTriesEnv)) {
+      this.numTries = numTriesEnv + 1;
+    }
+
+    const failedPublishPayloadsDirEnv = iso.getEnv(
+      "BRAINTRUST_FAILED_PUBLISH_PAYLOADS_DIR",
```

Comment:
should we set this by default to something, like in their home directory?

Reply:
Hmm I'm not sure if we want to have the SDK write stuff to a specific location on their filesystem by default. Isn't that part of the reason we had to remove `CACHE_PATH`? But if we're okay with asking people to turn it off explicitly then I guess we could.

Reply:
Let's keep it as is (opt-in) for now.